lib/plugin/plugin.c

```diff
@@ -380,31 +384,72 @@ struct bt_component_class *bt_plugin_find_component_class(
 	return comp_cls;
 }
 
-/* Allocate dirent as recommended by READDIR(3), NOTES on readdir_r */
+static struct {
+	pthread_mutex_t lock;
+	struct bt_plugin_set *plugin_set;
+	bt_bool recurse;
+} append_all_from_dir_info = {
+	.lock = PTHREAD_MUTEX_INITIALIZER
+};
+
 static
-struct dirent *alloc_dirent(const char *path)
+int nftw_append_all_from_dir(const char *file, const struct stat *sb, int flag,
```

Comment:
Not really a deal breaker, but I am wondering if the `fts` (`fts_open()`, `fts_read()`, etc.) family of functions are available on Windows?

It would be safer to use as currently, a plug-in that would use `nftw()` during its initialization would be problematic as `nftw()` is not guaranteed to be re-entrant.